cmd/queue/execprobe_test.go

```diff
@@ -106,49 +82,30 @@ func TestProbeShuttingDown(t *testing.T) {
 }
 
 func TestProbeQueueShuttingDownFailsFast(t *testing.T) {
-	ts := newProbeTestServer(func(w http.ResponseWriter) {
+	port := newProbeTestServer(t, func(w http.ResponseWriter) {
 		w.WriteHeader(http.StatusGone)
 	})
 
-	defer ts.Close()
-
-	u, err := url.Parse(ts.URL)
-	if err != nil {
-		t.Fatalf("%s is not a valid URL: %v", ts.URL, err)
-	}
-
-	port, err := strconv.Atoi(u.Port())
-	if err != nil {
-		t.Fatalf("Failed to convert port(%s) to int: %v", u.Port(), err)
-	}
-
 	start := time.Now()
-	if err = probeQueueHealthPath(1, port); err == nil {
+	if err := probeQueueHealthPath(1, port); err == nil {
 		t.Error("probeQueueHealthPath did not fail")
 	}
 
-	// if fails due to timeout and not cancelation, then it took too long
+	// if fails due to timeout and not cancelation, then it took too long.
 	if time.Since(start) >= 1*time.Second {
 		t.Error("took too long to fail")
 	}
 }
 
 func TestProbeQueueReady(t *testing.T) {
 	queueProbed := 0
-	ts := newProbeTestServer(func(w http.ResponseWriter) {
+	port := newProbeTestServer(t, func(w http.ResponseWriter) {
 		queueProbed++
```

Comment:
Same race potential.

Reply:
funny, had a bit of deja vu here :) https://github.com/knative/serving/pull/8311#discussion_r439669520

Reply:
tldr we actively removed the atomics bc the race detector didn't complain, turns out we were just getting lucky :(. I'm tempted to clean these up in a different PR though since it's not at all a new problem this introduced.

Reply:
Diff PR works for me.

Reply:
Follow-on PR to fix the races the race detector still doesn't detect here: https://github.com/knative/serving/pull/8900.